launcher/src/main/java/org/wildfly/core/launcher/CliCommandBuilder.java

```diff
@@ -376,8 +376,17 @@ public CliCommandBuilder addJavaOption(final String jvmArg) {
      */
     public CliCommandBuilder addJavaOptions(final String... javaOpts) {
         if (javaOpts != null) {
-            for (String javaOpt : javaOpts) {
-                addJavaOption(javaOpt);
+            for (int i = 0; i < javaOpts.length; i++) {
+                if (Arguments.isModuleSystemArgument(javaOpts[i])) {
+                    // handle java module system options
+                    if (i < javaOpts.length - 1) {
+                        this.javaOpts.add(Arguments.JavaModuleSystemArgument.of(javaOpts[i], javaOpts[++i]));
+                    } else {
+                        // ignore corrupted module system argument
```

Comment:
I don't think we should ignore these. We should just add the argument to the command and allow it to fail when executed.